pycbc/inference/models/__init__.py

```diff
@@ -242,9 +243,22 @@ def add_plugins(self):
         After this runs, ``self.retrieve_plugins`` is set to ``False``, so that
         subsequent calls to this will no re-add models.
         """
+
+
+        # Check if the Python version is 3.10 or newer
+        if sys.version_info >= (3, 10):
+            # Use the modern syntax with the 'group' keyword argument
+            def get_entry_points(group_name):
+                return entry_points(group=group_name)
+        else:
+            # Use the older syntax for Python 3.9 and earlier
+            all_entry_points = entry_points()
+            def get_entry_points(group_name):
+                return all_entry_points.get(group_name, ())
```

Comment:
Given that this is implemented identically twice, and that at some point we may want to drop the Python 3.9 hack, I suggest putting it into a small helper function in one of the generic modules. Also, can the version check decision happen inside `get_entry_points()`?